src/api/math.ts

```diff
@@ -248,3 +249,44 @@ export function calculateLiquidationData(
         liquidable: false,
     };
 }
+
+export function predictHealthFactor(args: PredictHealthFactorArgs): number {
+    const liquidationData = calculateLiquidationData(args.assetsConfig, args.assetsData, args.balances, args.prices)
+    const tokenHash = sha256Hash(args.tokenSymbol);
+    
+    const assetConfig = args.assetsConfig.get(tokenHash)!;
+    const assetPrice = args.prices.get(tokenHash)!;
+   
+    const calculations = calculateHealthBalance(args.balanceChangeType, assetConfig, assetPrice, args.amount);
+
+    const totalSupply = Number(liquidationData.totalLimit) + calculations.supply;
+    const totalBorrow = Number(liquidationData.totalDebt) + calculations.borrow;
+
+    if (Number(liquidationData.totalLimit) == 0) {
+        return 1;
+    }
+
+    return Math.min(Math.max(1 - totalBorrow / totalSupply, 0), 1);  // let's limit a result to zero below and one above
+}
+
+function calculateHealthBalance(changeType: BalanceChangeType, assetConfig: AssetConfig, price: bigint, value: bigint): {borrow:number, supply:number} {
```

Comment:
i suggest to merge two functions in one to keep consistency in math.ts file. one function - one calculation.